### ISSUE TITLE
Hera Support

### ISSUE THREAD

**Issue by akubaryk** [SWPC-IPE/WAM-IPE#325]
Today, Theia's replacement became available for a 30-day test period (during which Theia will be fully available with the exception of regular September maintenance). We will need to move eventually. Very initial support has been added in `ipe_refactor_no_netcdf` with 72e7c18, but no testing has been done on-machine.

**Comment by akubaryk** [comment]
After an email exchange with Ratko Vasic, the new DATADIR for Hera will be /scratch1/NCEPDEV/nems/emc.nemspara/RT/WAM-IPE/WAM-IPE_NEMS201606-20170302 -- this needs to be put into hera.config when the time comes. The machine detection code is getting unwieldy, and is also wrong in some places wrt Hera with 72e7c18, so I'd like to centralize and correct that as a part of this issue as well.

**Comment by IonospherePlasmasphereElectrodynamics** [comment]
@akubaryk 

since theia is down until tomorrow night, I wanted to try to run standalone ipe on hera. I could not find the suggested branch  "ipe_refactor_no_netcdf". 

anyway, when i tried the branch, "ipe_refactor_dev_wam", I got the following error. The main message is that we need netcdf 4.

;---
...
checking for NetCDF4 libraries... no
configure: WARNING:

Unable to locate NetCDF4 compilation helper script 'nc-config'.
Please specify --with-netcdf4=<LOCATION> as the full path prefix
where NetCDF4 has been installed.
NetCDF4 support is being disabled (equivalent to --with-netcdf4=no).

configure: error: Unable to find NetCDF4, we need NetCDF4.
make: *** No rule to make target `install'.  Stop.
;---

is NetCDF4 available on hera? 
any suggestion?

**Comment by IonospherePlasmasphereElectrodynamics** [comment]
i also tried to build wam-ipe. I got lots of errors. 

$ NEMS/NEMSAppBuilder rebuild app=wam-ipe


Major errors include,

IPE_MPI_Layer_Class.F90(4): error #7002: Error in opening the compiled module file.  Check INCLUDE paths.   [MPI]

...
Makefile:22: *** Environment variable ESMFMKFILE was not set..  Stop.
...
makefile:11: *** Error: !!!!!! Configure file is missing. Please create ./conf/configure.nems  by running ./configure !!!!!!.  Stop.

...
See /scratch2/NCEPDEV/stmp3/Naomi.Maruyama/WAM-IPE/log/appBuilder.NEMS.log.13424

**Comment by akubaryk** [comment]
Pretty sure I've sorted everything on Hera, and it'll go into #336 shortly -- there's a directory structure in place in /scratch1/NCEPDEV/swpc (data directory contains the files coming from WCOSS, while WAM-IPE_DATA directory contains further relevant subdirectories, initial conditions, IPE fix files, historical input parameters). That can be changed if the group wants something different.

The 1hr coupled cold start run on Hera completed in 4m13s (including all script setup, model/mediator initialization, etc.), which is really quite good. The restart hour completed in 4m35s. No verification work has been done yet, but we should do that soon.

Any jobs running on Theia will need the following setup in place of the current IC_DIR assignment:
```
export IC_DIR=/scratch1/NCEPDEV/swpc/WAM-IPE_DATA/ICS/$MODE/$CDATE
export IPE_IC_DIR=/scratch1/NCEPDEV/swpc/WAM-IPE_DATA/IPE_FIX
```

In the future I'd like to put the IPE_IC_DIR assignment in the machine configs, since it's going to be a fix folder now instead of containing the input states as well.

**Comment by wanghj2015** [comment]
@akubaryk any branch work on 'hera'? doesn't appear having config file for building on 'hera'?

**Comment by akubaryk** [comment]
Not sure what config file you're referring to but the merge branch in #336 has my Hera changes.

**Comment by wanghj2015** [comment]
NOTE: Configure using coupled mode.

NOTE: Command: ./configure coupled_intel_hera hera/wam-ipe

+ cp ../../conf/configure.nems.Hera.intel conf/configure.nems

cp: cannot stat ‘../../conf/configure.nems.Hera.intel’: No such file or
directory



Houjun Wang

**Comment by akubaryk** [comment]
Do another pull; I had forgotten to git add several new conf files for Hera. Should work now.

**Comment by IonospherePlasmasphereElectrodynamics** [comment]
@akubaryk 

I wanted to let you know that I am going to test the standalone ipe run on hera.

**Comment by IonospherePlasmasphereElectrodynamics** [comment]
building the standalone ipe failed. we need NetCDF4 as shown below.
@akubaryk , could you please let me know how i can solve the problem? thanks!

;-------------
checking for NetCDF4 libraries... no
configure: WARNING:

Unable to locate NetCDF4 compilation helper script 'nc-config'.
Please specify --with-netcdf4=<LOCATION> as the full path prefix
where NetCDF4 has been installed.
NetCDF4 support is being disabled (equivalent to --with-netcdf4=no).

configure: error: Unable to find NetCDF4, we need NetCDF4.
make: *** No rule to make target `install'.  Stop.

**Comment by akubaryk** [comment]
From IPELIB, what's the output of `source ../modulefiles/hera/wam-ipe`?

Also, can you put the full build log into a logfile, like so, and send me the path? The module load works fine for me on Hera, but there may be a difference in our environment.

`./ipe_build_from_scratch.sh > build.log 2>&1`

**Comment by IonospherePlasmasphereElectrodynamics** [comment]
@akubaryk 

thank you for your willingness to help.

$ source ../modulefiles/hera/wam-ipe



Currently Loaded Modules:
  1) intel/18.0.5.274   3) hdf5parallel/1.10.5   5) bacio/2.0.3   7) w3nco/2.0.6   9) nemsio/2.2.3
  2) impi/2018.0.4      4) netcdf/4.6.1          6) sp/2.0.2      8) w3emc/2.3.1  10) sigio/2.1.1


my path is 
/scratch2/NCEPDEV/stmp3/Naomi.Maruyama/tmp/issue325/WAM-IPE/IPELIB


here is the full log:
$ ./ipe_build_from_scratch.sh



make: *** No rule to make target `distclean'.  Stop.
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking for style of include used by make... GNU
checking dependency style of gcc... none
checking for ifort... no
checking for gfortran... gfortran
checking whether we are using the GNU Fortran compiler... yes
checking whether gfortran accepts -g... yes
checking for ifort... no
checking for gfortran... gfortran
checking whether we are using the GNU Fortran 77 compiler... yes
checking whether gfortran accepts -g... yes
checking for ar... ar
checking the archiver (ar) interface... ar
checking for ranlib... ranlib
checking for a sed that does not truncate output... /usr/bin/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking how to run the C preprocessor... gcc -E
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for nc-config... no
checking for NetCDF4 libraries... no
configure: WARNING:

Unable to locate NetCDF4 compilation helper script 'nc-config'.
Please specify --with-netcdf4=<LOCATION> as the full path prefix
where NetCDF4 has been installed.
NetCDF4 support is being disabled (equivalent to --with-netcdf4=no).

configure: error: Unable to find NetCDF4, we need NetCDF4.
make: *** No rule to make target `install'.  Stop.

**Comment by IonospherePlasmasphereElectrodynamics** [comment]
also the full log is saved 
/scratch2/NCEPDEV/stmp3/Naomi.Maruyama/tmp/issue325/WAM-IPE/IPELIB/build.log

**Comment by akubaryk** [comment]
It's looking in the /scratch4 nems path for the ESMF module.

My guess is that it's due to lines in your .cshrc or .bashrc that is loading that into your MODULEPATH for each login shell -- maybe try commenting out any `module use` statements in those files, depending on what shell you use, and then log out and log back in. You can force the issue on your current session with `module unuse /scratch4/NCEPDEV/nems/noscrub/emc.nemspara/soft/modulefiles` but that's obviously not a longterm solution.

**Comment by IonospherePlasmasphereElectrodynamics** [comment]
@akubaryk 

great! thanks a lot for your help! the new standalone ipe job is running on hera:
ROTDIR=/scratch1/NCEPDEV/stmp2/Naomi.Maruyama/issue325StandaloneIpeEmpE/

I will upload plots as soon as the runs are finished.

**Comment by IonospherePlasmasphereElectrodynamics** [comment]
another job is running with the self-consistent Electric field on hera:
/scratch1/NCEPDEV/stmp2/Naomi.Maruyama/issue325StandaloneIpeSE/

**Comment by wanghj2015** [comment]
Compare 1hr run differences between 'hera' and 'theia': zonal mean field differences ('hera' - 'theia'):

/scratch1/NCEPDEV/stmp2/Houjun.Wang/WAM-IPE_march_2013/
diff-IPE_Params.geo.201303160100-zm.nc4

And some figures from 'ncview' (not always good, but sometimes useful):
![Untitled presentation](https://user-images.githubusercontent.com/12502567/65466022-e9878700-de1a-11e9-8e80-a52b358ecf0c.png)

**Comment by wanghj2015** [comment]
@akubaryk Seems we are interested in looking at ion velocities too. I have added three variables to your 'post.py' (but I think we don't need 'he_plus_velocity' since it's just 'zero' in the original output file). Maybe you want to add the other two as the default in your 'post.py'?

    self.h_plus_velocity      = f.variables['h_plus_velocity'][:]
    self.he_plus_velocity     = f.variables['he_plus_velocity'][:]
    self.o_plus_velocity      = f.variables['o_plus_velocity'][:]

/scratch1/NCEPDEV/swpc/Houjun.Wang/github_clone/development_ipe_refactor_merge/IPELIB/scripts/post2.py

**Comment by akubaryk** [comment]
The people in charge of the NCEPLIBS modulefiles (silently) axed nemsio/2.2.3 and replaced it with nemsio/2.2.4 -- I committed this change and some more fixes specific to the standaloneGSM build to #336 just now; be sure to pull if you're trying to build on Hera.
